.github/workflows/publish-image.yml

```diff
@@ -0,0 +1,49 @@
+name: Docker Image Build and Publish
+
+on:
+  push:
+    branches: [ "main" ]
```

Comment:
Not sure if we should configure it to  push a new image on every merge to master. I think a better approach would be to run the job when a new tag is pushed.

Reply:
I'm fine either way, my intention was to have a latest image version bound to the head of the main branch. But I guess this can be adjusted in a followup PR once a decision is made in regards to the triggers

Reply:
Let's just trigger the release on push tags events:
```
on:
  push:
    tags:
      - 'v*'
```

Reply:
done